.github/pull_request_template.md

```diff
@@ -0,0 +1,10 @@
+<!--
+Checklist:
+ - Write / update unit tests
+ - Write / update integration (envtest) tests
+ - Ensure there is an issue for e2e tests if needed
+ - Make sure `make test` passes
+ - Test code changes manually
+ - Update readme and documentation
+ - Write PR description that highlights overall changes and add usage examples if applicable
+-->
```

Comment:
IMO we need an additional item that is to link the pull request to the original JIRA story and meanwhile the commit messages should also include JIRA story key.

Currently, one way to link is to prefix the PR subject with a JIRA story key. It can be introduced from commit message subject line if there is a single commit.